enterprise/frontend/src/metabase-enterprise/caching/components/InvalidateNowButton.tsx

```diff
@@ -45,6 +56,15 @@ const InvalidateNowFormBody = ({ targetName }: { targetName?: string }) => {
     [askConfirmation, targetName, submitForm],
   );
 
+  const buttonText = useMemo(() => {
+    const map: Record<ModelWithClearableCache, string> = {
+      dashboard: t`Clear cache for this dashboard`,
+      question: t`Clear cache for this question`,
+      database: t`Clear cache for this database`,
+    };
+    return map[targetModel];
```

Comment:
you also have ts-pattern if you want it!

Reply:
Yes! This is a good place for it. I'd originally avoided it here because the type didn't work well with it, but I've improved the types since then, and can now use `exhaustive` here

Reply:
You can also use `.otherwise()` to fallback to a base case when the types aren't quite there